lambdas/data-transfer/handler.py

```diff
@@ -47,14 +48,19 @@ def handler(event, context):
         # Check if the corresponding object exists in the target bucket
         try:
             target_s3.head_object(Bucket=TARGET_BUCKET, Key=target_key)
-        except ClientError:
-            # Not found
-            source_s3.download_file(bucket, f"{path}/{name}", tmp_filename)
-            target_s3.upload_file(tmp_filename, TARGET_BUCKET, target_key)
-            # Clean up the data
-            if os.path.exists(tmp_filename):
-                os.remove(tmp_filename)
-
-        object["data_url"] = target_url
+        except ClientError as ce:
+            try:
+                # Not found
+                source_s3.download_file(bucket, f"{path}/{name}", tmp_filename)
+                target_s3.upload_file(tmp_filename, TARGET_BUCKET, target_key)
+                # Clean up the data
+                if os.path.exists(tmp_filename):
+                    os.remove(tmp_filename)
+            except Exception as internal_exception:
+                print(f"Failed while trying to upload missing file at s3://{bucket}/{target_key}.")
+                traceback.print_exception(type(ce), ce, ce.__traceback__)
+                raise Exception(ce, internal_exception)
+
+        object["s3_filename"] = target_url
```

Comment:
To be clear, we want to change the key from `data_url` to `s3_filename`?  I'm not seeing this change referenced anywhere else in the code, where did this need come from?

Reply:
This was a bug in the code, when `data-transfer` lambda runs, the new url from the destination bucket should be used to build the stac item. The `buld-stac` lambda uses `s3_filename` to build the stac-item out of.